### PR TITLE
added glue metastore template and primary glue config

### PR DESCRIPTION
**Pull request by patduin**

### CODE REVIEW

File: templates.tf [hunk 1]
```diff
@@ -128,19 +128,40 @@ data "template_file" "ssh_metastores_yaml" {
...
     primary_metastore_latency          = var.primary_metastore_latency
+    primary_metastore_glue_account_id  = var.primary_metastore_glue_account_id
+    primary_metastore_glue_endpoint    = var.primary_metastore_glue_endpoint
     primary_metastore_whitelist        = join("", data.template_file.primary_metastore_whitelist.*.rendered)
     primary_metastore_mapped_databases = join("", data.template_file.primary_metastore_mapped_databases.*.rendered)
     local_metastores                   = join("", data.template_file.local_metastores_yaml.*.rendered)
     remote_metastores                  = join("", data.template_file.remote_metastores_yaml.*.rendered)
     remote_region_metastores           = join("", data.template_file.remote_region_metastores_yaml.*.rendered)
     ssh_metastores                     = join("", data.template_file.ssh_metastores_yaml.*.rendered)
+    glue_metastores                     = join("", data.template_file.glue_metastores_yaml.*.rendered)
```
Comment (barnharts4) [inline templates.tf:164]: can you fix this alignment?

File: variables.tf [hunk 2]
```diff
@@ -274,3 +293,15 @@ variable "primary_metastore_latency" {
...
+  type        = string
+  default     = ""
+  description = "IAM policy to attach to Waggle Dance Glue IAM role `waggle_dance_k8s_role_iam`."
+}
+
+variable "waggle_dance_glue_assume_role_policy" {
+  type        = string
+  default     = ""
+  description = "K8S assume role policy for Waggle Dance Glue IAM role `waggle_dance_k8s_role_iam`. K8S only"
+}
```
Comment (barnharts4) [inline variables.tf:307]: Maybe I'm missing something, but why do these two policy docs need to be passed in from the app?  They both seem like something that could be just in the module.

Comment (patduin) [reply variables.tf:307]: @barnharts4 is this ok for you?

Comment (barnharts4) [reply variables.tf:307]: I would rather we keep the actual IAM policies in the module, and pass in to the module *only* those things that need to change.  

- For `AssumeRole`, that should only be KIAM (or IRSA) role.  Here is an example of how we pass in only the KIAM role ARN to a module: https://github.com/ExpediaGroup/apiary-data-lake/blob/master/iam.tf#L43-L52
- For the `waggle_dance_glue_policy` policy document, this whole document should be in the module, and we should just pass in a variable to the module that is a list of resources that get added to the policy.  Here is an example of how we do something similar for the datalake IAM policies: https://github.com/ExpediaGroup/apiary-data-lake/blob/master/iam-policy-s3-buckets.tf#L92-L95

Comment (barnharts4) [reply variables.tf:307]: I'm happy to help collaborate on this too.

Comment (patduin) [reply variables.tf:307]: sure no problem, I'm gonna close this PR.

**Pull request closed without merging**